src/TextInput/index.js

```diff
@@ -85,60 +114,70 @@ class TextInput extends React.Component<Props, State> {
       prefix,
       onChange,
       inlineLabel,
+      suffix,
     } = this.props;
     const { focused } = this.state;
     return (
-      <View>
-        {label && !inlineLabel && (
-          <FormLabel filled={!!value} required={required} disabled={disabled}>
-            {label}
-          </FormLabel>
-        )}
-        <View
-          style={[
-            styles.inputContainer,
-            size === 'normal' ? styles.normalSize : styles.smallSize,
-            disabled
-              ? styles.inputContainerDisabled
-              : styles.inputContainerDefault,
-            focused
-              ? styles.inputContainerBorderFocused
-              : styles.inputContainerBorderDefault,
-          ]}
-        >
-          {prefix && <Prefix size={size}>{prefix}</Prefix>}
-          {label && inlineLabel && (
-            <InlineLabel>
-              <FormLabel
-                filled={!!value}
-                inlineLabel
-                required={required}
-                disabled={disabled}
-              >
-                {label}
-              </FormLabel>
-            </InlineLabel>
+      <TouchableWithoutFeedback onPress={this.focusTextInput}>
+        <View style={styles.inputWrapper}>
+          {label && !inlineLabel && (
+            <FormLabel filled={!!value} required={required} disabled={disabled}>
+              {label}
+            </FormLabel>
           )}
-          <RNTextInput
-            onFocus={this.onFocus}
-            onBlur={this.onBlur}
-            onChangeText={onChange}
-            placeholderTextColor={defaultTokens.colorPlaceholderInput}
-            editable={!disabled}
-            placeholder={placeholder}
-            value={value}
+          <View
             style={[
-              styles.inputField,
-              disabled ? styles.inputFieldDisabled : styles.inputFieldDefault,
+              styles.inputContainer,
+              size === 'normal' ? styles.normalSize : styles.smallSize,
+              disabled
+                ? styles.inputContainerDisabled
+                : styles.inputContainerDefault,
+              focused
+                ? styles.inputContainerBorderFocused
+                : styles.inputContainerBorderDefault,
             ]}
-          />
+          >
+            {prefix && <Prefix size={size}>{prefix}</Prefix>}
+            {label && inlineLabel && (
```

Comment:
both prefix and label will cause crash on android if they are 0 or empty string, please use prefix != null and label != null(after changing it to React.Node)